rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -326,9 +327,19 @@ void SequentialWriter::switch_to_next_storage()
   }
 
   storage_->update_metadata(metadata_);
+
+  // Check for overflow: if next_file_index_ is 0, we've wrapped around (very unlikely but possible)
+  if (next_file_index_ == 0) {
+    ROSBAG2_CPP_LOG_WARN_STREAM(
+      "File index counter has overflowed (wrapped to 0). "
+      "This should not happen in practice, but continuing with index 0. "
+      "If circular logging is enabled, ensure old files are deleted to avoid conflicts.");
+  }
+
   storage_options_.uri = format_storage_uri(
     base_folder_,
-    metadata_.relative_file_paths.size());
+    next_file_index_);
```

Comment:
This assignment would be better readable in one line
```suggestion
storage_options_.uri = format_storage_uri(base_folder_, next_file_index_);
```

Reply:
Updated.